### PR TITLE
test(stream): add state cleaning test

### PR DESCRIPTION
**Pull request by TennyZhuang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

As code

The test is passed, will add to CI in later PR.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by st1page — APPROVED** [review]
LGTM

**Review by fuyufjh — APPROVED** [review]

File: src/meta/Cargo.toml [hunk 1]
```diff
@@ -64,7 +64,7 @@ tokio = { version = "0.2", package = "madsim-tokio", features = [
     "signal",
 ] }
 tokio-retry = "0.3"
-tokio-stream = { version = "0.1", features = ["net"] }
+tokio-stream = { git = "https://github.com/madsim-rs/tokio.git", rev = "0c25710", features = ["net"] }
```
Comment (wangrunji0408) [inline src/meta/Cargo.toml:67]: Is this necessary? since we have patched this crate globally.

**Review by wangrunji0408 — APPROVED** [review]

**Comment by wangrunji0408** [comment]
> The test relies heavily on the use of sleep behavior. It is unclear whether a simulation test would result in faster performance.

I think it would help. I will add this to the simulation test later.

File: src/tests/state_cleaning_test/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# risingwave_state_cleaning_test
+
+The `risingwave_state_cleaning_test` crate has been designed specifically to test whether RisingWave can effectively clean outdated state records prior to reaching the watermark on time. Its functionality is described using TOML files, which specify the tests that should be executed. By utilizing this crate, developers can ensure that RisingWave is capable of properly managing state records, thereby improving overall application performance and providing a more reliable end-user experience.
```
Comment (xxchan) [inline src/tests/state_cleaning_test/README.md:3]: Is this written by GPT? 🥸

**Pull request merged**